proto/substrait/algebra.proto

```diff
@@ -810,6 +810,7 @@ message Expression {
       int64 time = 17;
       IntervalYearToMonth interval_year_to_month = 19;
       IntervalDayToSecond interval_day_to_second = 20;
+      IntervalMonthToNano interval_month_to_nano = 36;
```

Comment:
I aligned with existing names, but the `To` feels weird to me - I guess it indicates the literal contains data for precisions from X to Y, but I feel these would be clearer without it. Can't rename the existing ones w/o breaking things, but should this be `IntervalMonthDayNano` ?